src/emu/diimage.cpp

```diff
@@ -1309,10 +1309,9 @@ void device_image_interface::update_names()
 			count++;
 	}
 
-	m_canonical_instance_name = string_format("%s%d", inst_name, index + 1);
 	if (count > 1)
 	{
-		m_instance_name = m_canonical_instance_name;
```

Comment:
Furthermore, if you had `promchip`/`prom` and `promimage`/`prom` and you didn't want this to fail a validity check, wouldn't the right thing to do be to drop the briefinstance name?  Do you really want one of the above to be arbitrarily assigned `prom1-4` and another be assigned to `prom5-8`?